### ISSUE TITLE
Setup JRuby extension

### ISSUE THREAD

**Issue by yorickpeterse** [YorickPeterse/oga#16]
Currently the `native-ext` branch contains a C extension for the XML lexer. This same setup should be ported over to JRuby. Because I don't want to maintain the same Ragel grammar in 2 places I'll be generating JRuby/C code using this grammar. The following steps are involved:
- [x] Port over the C extension to a JRuby extension, keep the grammars separate for now. Once done, test everything to make sure it's working.
- [x] Port over the setup to some kind of "template" system that generates the C and JRuby code. This should remove the need for 2 identical Ragel grammars.

Basically the Ragel grammar would contain a set of placeholders that will be replaced with either C or Java code. The resulting file is then passed to Ragel and compiled.
